modules/images/webp-uploads/load.php

```diff
@@ -683,3 +683,91 @@ function webp_uploads_update_rest_attachment( WP_REST_Response $response, WP_Pos
+ * Updates webp image when original image is edited
+ *
+ * @since n.e.x.t
+ *
+ * @param bool|null       $override  Value to return instead of saving. Default null.
+ * @param string          $file      Name of the file to be saved.
+ * @param WP_Image_Editor $image     The image editor instance.
+ * @param string          $mime_type The mime type of the image.
+ * @param int             $post_id   Attachment post ID.
+ *
+ * @return bool|null
+ */
+function webp_uploads_update_image_onchange( $override, $file, $image, $mime_type, $post_id ) {
+	if ( $override !== null ) {
+		return $override;
+	}
+
+	$valid_mime_transforms      = webp_uploads_get_upload_image_mime_transforms();
+	$original_directory         = pathinfo( $file, PATHINFO_DIRNAME );
+	$filename                   = pathinfo( $file, PATHINFO_FILENAME );
+	$sizes                      = get_intermediate_image_sizes();
+	$_wp_additional_image_sizes = wp_get_additional_image_sizes();
+	$target                     = ! empty( $_REQUEST['target'] ) ? preg_replace( '/[^a-z0-9_-]+/i', '', $_REQUEST['target'] ) : '';
+	$nocrop                     = false;
+	$current_mime_type          = get_post_mime_type( $post_id );
+	$allowed_mimes              = array_flip( wp_get_mime_types() );
+
+	foreach ( $valid_mime_transforms[ $mime_type ] as $targeted_mime ) {
+		if ( $targeted_mime === $current_mime_type ) {
+			continue;
+		}
+
+		if ( ! isset( $allowed_mimes[ $targeted_mime ] ) || ! is_string( $allowed_mimes[ $targeted_mime ] ) ) {
+			return new WP_Error( 'image_mime_type_invalid', __( 'The provided mime type is not allowed.', 'performance-lab' ) );
+		}
+
+		if ( ! wp_image_editor_supports( array( 'mime_type' => $targeted_mime ) ) ) {
+			return new WP_Error( 'image_mime_type_not_supported', __( 'The provided mime type is not supported.', 'performance-lab' ) );
+		}
+
+		$extension   = explode( '|', $allowed_mimes[ $targeted_mime ] );
+		$destination = trailingslashit( $original_directory ) . "{$filename}.{$extension[0]}";
+
+		$image->save( $destination, $targeted_mime );
+
+		$new_image = wp_get_image_editor( $destination );
+
+		if ( is_wp_error( $new_image ) ) {
+			return $new_image;
+		}
+
+		if ( 'thumbnail' === $target ) {
+			$nocrop = true;
+		}
+
+		if ( ! isset( $sizes ) ) {
+		  continue;
+		}
+
+		$_sizes = array();
+
+		foreach ( $sizes as $size ) {
+
+			if ( isset( $_wp_additional_image_sizes[ $size ] ) ) {
+				$width  = (int) $_wp_additional_image_sizes[ $size ]['width'];
+				$height = (int) $_wp_additional_image_sizes[ $size ]['height'];
+				$crop   = ( $nocrop ) ? false : $_wp_additional_image_sizes[ $size ]['crop'];
+			} else {
+				$height = (int) get_option( "{$size}_size_h" );
+				$width  = (int) get_option( "{$size}_size_w" );
+				$crop   = ( $nocrop ) ? false : get_option( "{$size}_crop" );
+			}
+
+			$_sizes[ $size ] = array(
+				'width'  => $width,
+				'height' => $height,
+				'crop'   => $crop,
+			);
+		}
+
+		$new_image->multi_resize( $_sizes );
+	}
```

Comment:
As @mitogh mentioned, the one critical thing that is missing here is to update the metadata with the new sources.

You will need to store the return value from the above `save()` method call under the root `$metadata['sources']`, and you will need to store the return values from the above `multi_resize()` method call under each respective `$metadata[ $size ]['sources']`.

Since this hook cannot modify those, I'd say it would be best to add a separate `add_filter` for `wp_update_attachment_metadata` _within_ this method, so that it only gets hooked in once we've created these images. For example:
```php
add_filter(
	'wp_update_attachment_metadata',
	function( $metadata, $meta_post_id } use ( $post_id, $orig_source, $size_sources ) { // Reuse the same values for the new images generated above.
		if ( $meta_post_id !== $post_id ) { // Check to ensure we're working with the right post.
			return $metadata:
		}
		// New function to add, which expands/updates the given metadata with the additional sources passed.
		return webp_uploads_update_sources( $metadata, $orig_source, $size_sources );
	},
	10,
	2
);
```

Then, in the separate hook callback, update the metadata accordingly before it is saved.

Reply:
@felixarntz Actually I was able to figure out the adding `wp_update_attachment_metadata` filter part myself, but I was still looking for an optimal way to update metadata. Thankfully your in this PR made the code much simpler and so I was able to finish the rest of updating metadata part too!